pkg/reconcile/pipeline/infinispan/handler/provision/statefulsets.go

```diff
@@ -112,9 +117,10 @@ func ClusterStatefulSetSpec(statefulSetName string, i *ispnv1.Infinispan, ctx pi
 					Annotations: annotationsForPod,
 				},
 				Spec: corev1.PodSpec{
-					Affinity:                  i.Affinity(),
-					Tolerations:               i.Tolerations(),
-					TopologySpreadConstraints: i.TopologySpreadConstraints(),
+					TerminationGracePeriodSeconds: &gracePeriod,
```

Comment:
`TerminationGracePeriodSeconds` in the PodSpec is a int64 pointer the same as in our types, so there's no need to explicitly check this value is set we can just pass the return value of the `types_util.go` method directly.

It's better for us to leverage the default `TerminationGracePeriodSeconds` determined by Kubernetes, instead of explicitly defining 30s in our code, as different k8s vendors may change this value or the value may be changed in future k8s versions.